src/axom/core/ArrayView.hpp

```diff
@@ -311,6 +379,47 @@ ArrayView<T, DIM, SPACE>::ArrayView(
 #endif
 }
 
+//------------------------------------------------------------------------------
+template <typename T, int DIM, MemorySpace SPACE>
+AXOM_HOST_DEVICE ArrayView<T, DIM, SPACE> ArrayView<T, DIM, SPACE>::subspan(
+  const StackArray<IndexType, DIM>& offsets,
+  const StackArray<IndexType, DIM>& counts)
+{
+#ifdef AXOM_DEBUG
+  for(int dim = 0; dim < DIM; dim++)
+  {
+    assert(offsets[dim] >= 0 && offsets[dim] < this->shape()[dim]);
+    if(counts[dim] >= 0)
+    {
+      assert(offsets[dim] + counts[dim] <= this->shape()[dim]);
```

Comment:
Use `SLIC_ERROR_IF` instead of `assert`?

Reply:
I think that is better, but others should weigh in.

Reply:
I think the problem is that Array and ArrayView reside in the core component, so it might be complicated to depend on SLIC.

Reply:
@publixsubfan that's a good point. Leave as is for now.

Reply:
Thanks @publixsubfan -- yes, `slic` depends on `core`, so we can't use its macros in `core`.